xapian-bindings/configure.ac

```diff
@@ -94,6 +94,9 @@ XO_LIB_XAPIAN([], [],
 			  [^\([0-9]*\.[0-9]*[13579]\)\..*$], [-\1]))
 XO_REQUIRE([1.4.0])
 
+dnl Check for xapian-letor.
+XO_LIB_XAPIANLETOR([], [], [xapianletor-config])
```

Comment:
I think we need a different `ACTION-IF-NOT-FOUND` to the default, because we don't want bindings to not build when letor isn't present; we just want a notification (probably `AC_MSG_WARN`) that we aren't going to build the letor bindings, and then we keep going — various things (principally in `Makefile.am`) would then need to be conditional so we don't try to build the letor bindings in the absence of the underlying library.

(Currently, it looks like `XO_LIB_XAPIANLETOR` supports `ACTION-IF-FOUND` but not `ACTION-IF-NOT-FOUND`, so that would need implementing.)

This _could_ be left to a separate PR, but it would mean that the bindings aren't fully releasable until done.

Reply:
sure I would create a new PR for that after this is merged

Reply:
> (Currently, it looks like `XO_LIB_XAPIANLETOR` supports `ACTION-IF-FOUND` but not `ACTION-IF-NOT-FOUND`, so that would need implementing.)

Hmm, it documents support but comparing with `xapian-1.5.m4` it seems the equivalent of `if test -z "$XAPIAN_CONFIG"; then` ... has been omitted for some reason.

That really needs fixing, and I think it makes much more sense to sort this out before we merge this patch.  Otherwise until this gets addressed xapian-letor becomes a hard requirement for xapian-bindings for anybody working on git master, which really isn't helpful.

Reply:
I have fixed all those issues in this PR.

Reply:
@jaylett Are you happy the changes you requested have been made?